simcon_project/users/views/student_management.py

```diff
@@ -0,0 +1,123 @@
+                        first_name = ""
+                        last_name = ""
+                        password = ""
+                        user = Student.objects.create(email=email, first_name=first_name, last_name=last_name, password=password, added_by=added_by, )
+                        user.set_unusable_password()
+
+                        #adds student to current label and/or the All Students label
+                        label = SubjectLabel.objects.get(label_name=name, researcher=added_by)
+                        label.students.add(user)
+                        if not name == "All Students":
+                            label = SubjectLabel.objects.get(label_name="All Students", researcher=added_by)
+                            label.students.add(user)
+
+                        #collects the current domain of the website and the users uid
+                        current_site = get_current_site(request)
+                        site = current_site.domain
+                        uid = urlsafe_base64_encode(force_bytes(user.pk))
+
+                        #creates the subject and message content for the emails
+                        subject = 'Activate Your Simulated conversations account'
+                        message = 'Hi, \nPlease register here: \nhttp://' + site + '/user-registration/'\
+                              + uid + '\n'
+
+                        #sends the email
+                        send_mail(subject, message, 'simulated.conversation@mail.com', [email], fail_silently=False)
+                    else:
+                        messages.error(request, 'Student already exists',
+                                       fail_silently=False)
+
+                else:
+                    # checks to make sure user exists
+                    if Student.objects.filter(email=email):
+
+                        # adds student to current label
+                        user = Student.objects.get(email=email)
+                        label = SubjectLabel.objects.get(label_name=name, researcher=added_by)
+                        label.students.add(user)
+
+                    else:
+                        messages.error(request, 'Student does not already exist',
+                                       fail_silently=False)
+        if request.POST.get('label_name'):  #create new label
+            savefldr = NewLabel(request.POST)
+            if savefldr.is_valid():
+                label_name = savefldr.cleaned_data.get("label_name")
+
+                #if the label does not already exist, create it
+                if not SubjectLabel.objects.filter(label_name=label_name, researcher=added_by):
+                    SubjectLabel().create_label(label_name, added_by)
+                else:
+                    messages.error(request, 'Label name already exists',
+                                   fail_silently=False)
+
```

Comment:
You can get around the page refresh issue if you move lines 27-56 to just before the return statement. So the if statement checking for POST method should be at the top and everything else happens after. That way all your tables will generate with the newly added students/labels. Also, and maybe this doesn't need to happen this ticket but I think it could useful to break this up into separate functions just for better readability.